internal/api/handlers/requestadapters/schedulers.go

```diff
@@ -58,14 +61,53 @@ func FromApiPatchSchedulerRequestToChangeMap(request *api.PatchSchedulerRequest)
 		patchMap[patch_scheduler.LabelSchedulerRoomsReplicas] = int(request.GetRoomsReplicas())
 	}
 
+	if request.Autoscaling != nil {
+		newAutoscaling, err := fromApiOptionalAutoscalingToChangeMap(request.GetAutoscaling())
+		if err != nil {
+			return map[string]interface{}{}, err
+		}
+		patchMap[patch_scheduler.LabelAutoscaling] = newAutoscaling
+	}
+
 	if request.Forwarders != nil {
 		patchMap[patch_scheduler.LabelSchedulerForwarders] = fromApiForwarders(request.GetForwarders())
 	}
 
-	return patchMap
+	return patchMap, nil
+}
+
+func fromApiOptionalAutoscalingToChangeMap(apiAutoscaling *api.OptionalAutoscaling) (map[string]interface{}, error) {
+	changeMap := make(map[string]interface{})
+
+	if apiAutoscaling.Enabled != nil {
+		changeMap[patch_scheduler.LabelAutoscalingEnabled] = apiAutoscaling.GetEnabled()
+	}
+
+	if apiAutoscaling.Min != nil {
+		changeMap[patch_scheduler.LabelAutoscalingMin] = apiAutoscaling.GetMin()
+	}
+
+	if apiAutoscaling.Max != nil {
+		changeMap[patch_scheduler.LabelAutoscalingMax] = apiAutoscaling.GetMax()
+	}
+
+	if apiAutoscaling.Policy != nil {
+		autoscalingPolicy, err := fromApiAutoscalingPolicy(apiAutoscaling.GetPolicy())
+		if err != nil {
+			return map[string]interface{}{}, err
+		}
+		changeMap[patch_scheduler.LabelAutoscalingPolicy] = autoscalingPolicy
+	}
+
+	return changeMap, nil
 }
 
 func FromApiCreateSchedulerRequestToEntity(request *api.CreateSchedulerRequest) (*entities.Scheduler, error) {
+	schedulerAutoscaling, err := fromApiAutoscaling(request.GetAutoscaling())
```

Comment:
In this case I think that Autoscaling could be nil, right?

Reply:
Yes... Inside the fromApiAutoscaling method, we check if it is nil